src/transforms/remap.rs

```diff
@@ -87,22 +87,27 @@ impl FunctionTransform for Remap {
             None
         };
 
+        let mut target: VrlTarget = event.into();
+
         let mut runtime = Runtime::default();
 
-        let result = match event {
-            Event::Log(ref mut event) => runtime.resolve(event, &self.program),
-            Event::Metric(ref mut event) => runtime.resolve(event, &self.program),
-        };
+        let result = runtime.resolve(&mut target, &self.program);
 
         match result {
-            Ok(_) => output.push(event),
+            Ok(_) => {
+                for event in target.into_events() {
+                    output.push(event)
+                }
+            }
             Err(Terminate::Abort) => {
                 emit!(RemapMappingAbort {
                     event_dropped: self.drop_on_abort,
                 });
 
                 if !self.drop_on_abort {
-                    output.push(original_event.unwrap_or(event))
+                    if let Some(event) = original_event {
+                        output.push(event)
+                    }
```

Comment:
Hmm, doesn't this change the original behavior, given that we no longer default to using `event`?

Reply:
Apologies, I meant to call this change out as it is subtle. The original behavior using `unwrap_or` would never actually call the `unwrap_or` if it got this far as if `drop_on_abort` is `false`, and the program _can_ abort, then `original_event` is always `Some`. If `drop_on_abort` is `false` but the program _cannot_ abort, then this `Err(Terminate::Abort)` match arm will never be hit.

It would be nice to model this a bit more clearly. I'll see if I can think of something.

Reply:
I couldn't come up with anything better at the moment so I just swapped the `if let Some`s for `expect` to at least make the invariant more clear.

Reply:
Ah yep, good catch. The `expect` solution seems fine to me.